gammapy/modeling/models/core.py

```diff
@@ -428,6 +430,18 @@ def read(cls, filename):
     def from_yaml(cls, yaml_str, path=""):
         """Create from YAML string."""
         data = yaml.safe_load(yaml_str)
+        if "metadata" in data.keys():
```

Comment:
This meta information will be created only if a model is read from disc? Just 

```
m = PowerLawSpectralModel()
sk = SkyModel(spectral_model=m, name="model")
sk.to_dict()["metadata"]
```
is `None` ?